### PR TITLE
feat: add tab name and portlet info to analytics events

### PR DESCRIPTION
**Pull request by bjagg**
##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]
-   [ ] tests are included
-   [ ] documentation is changed or added
-   [ ] new security keys and comments added to `security.properties`
-   [ ] new general keys and comments added to `portal.properties`
-   [ ] any changes that impact configuration or the database DDL added to `CHANGES.md`
-   [ ] [message properties][] have been updated with new phrases
-   [x] view conforms with [WCAG 2.0 AA][]

##### Description of change

Add tab (name) and portlet info (ID and title) to analytics events.

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

File: uPortal-webapp/src/main/webapp/scripts/portal-analytics.js [hunk 1]
```diff
@@ -42,9 +42,16 @@
             return;
         }
 
+        const tab = document.querySelector('li.fl-tabs-active span');
```
Comment (ChristianMurphy) [inline uPortal-webapp/src/main/webapp/scripts/portal-analytics.js:45]: Really like seeing `const` here, I vastly prefer it over `var`.
Just to double check, what is the current browser support matrix?
Every evergreen browser does support `const` and has for a while, IE still does not.

**Review by ChristianMurphy — APPROVED** [review]
Thanks @bjagg! looks good

File: uPortal-webapp/src/main/webapp/scripts/portal-analytics.js (hunk 1 above)
Comment (bjagg) [reply uPortal-webapp/src/main/webapp/scripts/portal-analytics.js:45]: We no longer support IE, only Edge. 👏🏽

Comment (ChristianMurphy) [inline uPortal-webapp/src/main/webapp/scripts/portal-analytics.js:45]: It looks like ESLint is unhappy https://github.com/uPortal-Project/uPortal/actions/runs/5954783612/job/16152048170?pr=2694#step:6:221

This is likely due to a preset defaulting to an older version of JavaScript
Try setting `env` with` `browser` and `es2023` https://eslint.org/docs/latest/use/configure/language-options

https://github.com/uPortal-Project/uPortal/blob/15de05e2d7bee8a55d20f0560dc23c0e9a801025/package.json#L26C4-L60

**Review by ChristianMurphy — CHANGES_REQUESTED** [review]
It looks like ESLint may need a configuration update

**Review by ChristianMurphy — APPROVED** [review]

**Review by cbeach47 — COMMENTED** [review]
@bjagg - Fixed the package lock conflict and confirmed the new details are being sent in the request locally.

Seems a bit odd that the `tab_name` is scraping the text content.  It's ending up with values such as `tab_name: "TestingEdit text Testing"`. If this is as-intended, then 👍 

I did notice that the raw event being captured does _not_ include the tab name and portlet info. I have the `events.analytics.log.level` set to `AUTHENTICATED`, and was logged in as an admin. Was this as designed? Currently this is the type of data being saved to the raw events table for `event_data`:
```
{
  "@c": ".AnalyticsPortalEvent",
  "timestamp": 1698962129046,
  "serverId": "...",
  "eventSessionId": "m4j5hxy3",
  "userName": "admin",
  "user": "admin",
  "eventDate": 1698962129046,
  "type": "link",
  "url": "http://localhost:8080/uPortal/f/testing/p/test-portlet-2.u22l1n16/normal/action.uP?_csrf=a5d65dfd-0617-4907-a53c-80f149d158ec&pP_testId=2"
}
```

**Comment by cbeach47** [comment]
Oh I see now - https://github.com/uPortal-Project/uPortal/pull/2695/ deals with the persistence of the new data.

**Review by cbeach47 — APPROVED** [review]

**Pull request merged**